core/lib/dal/src/consensus_dal.rs

```diff
@@ -457,9 +444,10 @@ impl ConsensusDal<'_, '_> {
         .fetch_one(self.storage)
         .await?;
 
-        Ok(row
-            .number
-            .map(|number| attester::BatchNumber(number as u64)))
+        let Some(n) = row.number else { return Ok(None) };
+        Ok(Some(attester::BatchNumber(
+            n.try_into().context("overflow")?,
+        )))
```

Comment:
This looks difficult to read. NB the previous pattern is [used all over blocks_dal](https://github.com/matter-labs/zksync-era/blob/46cfdd8f7f2d8003a91355ee4c76ab6836d3c0d7/core/lib/dal/src/blocks_dal.rs#L115), with `u32` at that. I'd say it's impossible that we'll overflow a `u64` with minute or even second level batches, since it can handle milliseconds since unix epoch.

Reply:
The "as" pattern is a silent wrap, not many people realize that and that's dangerous. I don't see what you find unreadable about try_into() construct - it does exactly what it says: try to cast one integer type to the other and raise an error if it doesn't fit. LMK if you know a more readable way to express that.

Reply:
Not the `.try_into` itself, but the whole thing with the `let Some(n)` having a `return` on the same line and then another `Ok(Some(` is a bit busy.

I understand it's dangerous in general, however this data is coming from the database and we know that the rest of the program is casting it to be a `u32`, while we have `u64` here. Maybe wrap it up into some conversion function and move it out of here?

Reply:
added more lines

Reply:
It is true that at the moment this is the only place it's used in `consenus_dal`, but this kind of conversion exists in numerous places in `blocks_dal`. I thought in the interest of consistency you could add safe conversions so next time it's the obvious choice.

Reply:
I'll update the blocks_dal occurrences in a separate PR